v2/datastream-to-spanner/src/test/java/com/google/cloud/teleport/v2/templates/DataStreamToSpannerITBase.java

```diff
@@ -145,54 +143,68 @@ protected CheckResult check() {
   }
 
   /**
-   * Performs the following steps: Uploads session file to GCS. Created schema in Spanner by reading
-   * DDL from spanner-schema.sql file. Creates Pubsub resources. Launches DataStreamToSpanner
-   * dataflow job.
+   * Performs the following steps: Uploads session file to GCS. Creates Pubsub resources. Launches
+   * DataStreamToSpanner dataflow job.
    *
-   * @param testName will be used as postfix in generated resource ids
+   * @param identifierSuffix will be used as postfix in generated resource ids
    * @param sessionFileResourceName Session file name with path relative to resources directory
-   * @param spannerSchemaResourceName spanner schema file name with path relative to resources
-   *     directory
+   * @param transformationContextFileResourceName Transformation context file name with path
+   *     relative to resources directory
+   * @param gcsPathPrefix Prefix directory name for this DF job. Data and DLQ directories will be
+   *     created under this prefix.
    * @return dataflow jobInfo object
    * @throws IOException
    */
   protected LaunchInfo launchDataflowJob(
-      String testName,
+      String identifierSuffix,
       String sessionFileResourceName,
-      String spannerSchemaResourceName,
+      String transformationContextFileResourceName,
+      String gcsPathPrefix,
       SpannerResourceManager spannerResourceManager,
       PubsubResourceManager pubsubResourceManager,
       Map<String, String> jobParameters)
       throws IOException {
```

Comment:
These type of changes are exactly what I'm a bit concerned about with having these wrappers in an  `*ITBase` class. This will affect all the ITs which use this class.
At the very least, we should refactor this to a `Builder` so that adding new fields to the builder does not lead to ripple code changes across all ITs.

Reply:
I tried to reuse the same method for both Sharded and non sharded tests. That's why a slight modification was required. Otherwise wouldn't have required to change this method.